mypy/constraints.py

```diff
@@ -172,6 +172,10 @@ def infer_constraints_if_possible(template: Type, actual: Type,
     if (direction == SUPERTYPE_OF and
             not mypy.subtypes.is_subtype(actual, erase_typevars(template))):
         return None
+    if (direction == SUPERTYPE_OF and isinstance(template, TypeVarType) and
+            not mypy.subtypes.is_subtype(actual, erase_typevars(template.upper_bound))):
+        # This is not caught by the above branch because of the erase_typevars() call.
```

Comment:
This comment was initially confusing. It wasn't clear to which `erase_typevars()` call this refers to. Maybe make it more explicit, such as saying that `erase_typevars` would return `Any` for a type variable.